martide/literature
# feat(WEB-8917): [Literature] Unable to upload images

This asynchronously stores different sizes of images in post/publication/authors/tags instead of doing it synchronously when `cast_attachments/3` is called. This will prevent the blocking when saving the said resources especially when uploading huge files.

To test:

Note that all tests below should be done with the master branch of https://github.com/martide/elixir

1. Since this change hasn't been released yet. In the `elixir` repository code, replace all written `{:literature, "~> 0.1"},` with `{:literature, git: "https://github.com/martide/literature.git", branch: "WEB-8917-bg-job-for-uploading-converted-img"},` Use find and replace in your code editor
2. Run `mix deps.get` in your terminal
3. Run the server `mix phx.server`
4. Go to `https://localhost:4103/literature/blog/posts` then create a new post
5. In the feature image and any other images, upload a huge file. You may try 2 - 8 MB (larger than 8MB will display a `too large file` error)
6. You should be able to save the post
7. Go to https://localhost:4203/en/blog
8. If you inspect the page, you should be able to see a set of images with different sizes in the srcset attribute. You should be able to successfully access and download each link in the srcset. For example if `srcset="https://some-link-1.jpg 100w, https://some-link-2.jpg 200w, ..."` then you should be able to access and download through links `https://some-link-1.jpg`, `https://some-link-2.jpg`, etc... Sample below:
![image](https://github.com/martide/literature/assets/22972528/65466b1c-410b-42ef-b21b-8ef37e0d5f41)
